### PR TITLE
Fix walletconnect connections don't disconnect automatically when dApp disconnects

### LINKED ISSUE #1930: WalletConnect connections don't disconnect automatically when dApp disconnects
### Expected behavior
Once a wallet connection is done, and the connected dApp disconnects from Lisk Wallet, the app should automatically remove the dApp connection from the external connections list.

### Actual behavior
When the dApp disconnects from Lisk, the connection stills appearing on the wallet app.

### Steps to reproduce
1. Establish a new WalletConnect connection from a dApp.
2. Disconnect from the dApp (not from Lisk Wallet).
3. You will see that the connection stills appearing on Lisk Wallet, when was already closed by the dApp.

### Which version(s) does this affect? (Environment, OS, etc...)
3.0.0.

### PR DESCRIPTION
**Pull request by ClementeSerrano**
### What was the problem?

This PR resolves #1930

### How was it solved?

- [x] Refactor WC library implementation to be in sync with Lisk Desktop implementation.
- [x] Adapt components to new WC refactor.
- [x] Fix app display name rendering on connection components.
- [x] Fix app breaking on unsupported `sign_message` request. 

### How was it tested?

- [x] iOS Simulator.
- [x] Android Emulator.

### CODE REVIEW

File: libs/wcm/constants/permissions.js [hunk 1]
```diff
@@ -1,9 +1,4 @@
 export const SIGNING_METHODS = {
-  SIGN_TRANSACTION: { key: 'sign_transaction', title: 'Sign transaction' },
+  SIGN_TRANSACTION: { key: 'sign_transaction', title: 'Signature request' },
```
[the remark this answers is not in the document]
Comment (ClementeSerrano) [reply libs/wcm/constants/permissions.js:2]: also not translated from desktop. Im trying to keep the implementation the same as possible

File: libs/wcm/context/ConnectionEventsManagerWrapper.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  }, []);
+
+  useEffect(() => {
+    if (signClient?.on) {
+      Object.keys(EVENTS).forEach((eventName) => {
+        signClient.on(EVENTS[eventName], eventHandler.bind(null, EVENTS[eventName]));
+      });
+    } else {
+      // eslint-disable-next-line no-console
+      console.log('There was an error initializing the client');
```
[the remark this answers is not in the document]
Comment (ClementeSerrano) [reply libs/wcm/context/ConnectionEventsManagerWrapper.js:38]: hmmm not sure what to do here. How should we handle this on the UI?

File: libs/wcm/constants/permissions.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ClementeSerrano) [reply libs/wcm/constants/permissions.js:2]: planning to address on https://github.com/LiskHQ/lisk-mobile/issues/2023 as discussed.

File: libs/wcm/context/ConnectionEventsManagerWrapper.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ClementeSerrano) [reply libs/wcm/context/ConnectionEventsManagerWrapper.js:38]: planning to address on https://github.com/LiskHQ/lisk-mobile/issues/2023 as discussed.

File: libs/wcm/hooks/useSession.js [hunk 3]
```diff
@@ -84,22 +95,48 @@ const useSession = () => {
...
+
   useEffect(() => {
-    if (signClient?.session && !session.loaded) {
-      const lastKeyIndex = signClient.session.keys.length - 1;
-      const data =
-        lastKeyIndex === 0 ? signClient.session.get(signClient.session.keys[lastKeyIndex]) : false;
-      setSession({ ...session, loaded: true, data });
+    if (signClient?.session && !hasLoaded) {
+      (async () => {
+        await loadSessions();
```
[the remark this answers is not in the document]
Comment (ClementeSerrano) [reply libs/wcm/hooks/useSession.js:159]: planning to address on https://github.com/LiskHQ/lisk-mobile/issues/2023 as discussed.

**Review by ManuGowda — APPROVED** [review]

**Pull request merged**